pkgs/development/libraries/libgtop/default.nix

```diff
@@ -26,6 +27,7 @@ stdenv.mkDerivation rec {
     perl
     gettext
     gobject-introspection
+    fake-uname
```

Comment:
How is it ensured that this uname takes precedence over other uname implementations?

Reply:
https://github.com/NixOS/nixpkgs/blob/6abb95c34492ccdbee0f6595b07f798e7bb11e0f/pkgs/stdenv/linux/default.nix#L492-L493

https://github.com/NixOS/nixpkgs/blob/6abb95c34492ccdbee0f6595b07f798e7bb11e0f/pkgs/stdenv/generic/default.nix#L133-L134

https://github.com/NixOS/nixpkgs/blob/6abb95c34492ccdbee0f6595b07f798e7bb11e0f/pkgs/stdenv/generic/setup.sh#L274-L287

somehow PATH contains the nativeBuildInputs first and then common-path
```
declare -x PATH="/nix/store/k2vd7w2ixbqbgd9v88am8780111bng7j-pkg-config-wrapper-0.29.2/bin:/nix/store/lw5mv8qzyalks7w80a6615na48i8lny1-gtk-doc-1.33.2/bin:/nix/store/n48i43wkly6kfhxz4ypcxbsv2pr7ggzh-perl-5.36.0/bin:/nix/store/zms41v0fgzfq4z629kn5jrlcv7hf9w7v-gettext-0.21/bin:/nix/store/3qgzjk0i8g7gif5jly94nbph0jwx4d0s-gobject-introspection-wrapped-1.74.0-dev/bin:/nix/store/02ggkplpkvdjkr20bylnidhn7cyh66m4-glib-2.74.3-dev/bin:/nix/store/k886wb6j8m1jwxf5pf1d5mdmlyblyd1y-glib-2.74.3-bin/bin:/nix/store/g2nfbpind37b3v3y6zbszmcgfmrj5v7x-patchelf-0.15.0/bin:/nix/store/wn31i3dzwahz6ccws8bs1nwyqrpgsvj7-gcc-wrapper-11.3.0/bin:/nix/store/sxdx80lmk4zkhb51f4x5dgqvxgmx55wl-gcc-11.3.0/bin:/nix/store/s7ip867mrpnnjlppbnxlcsq10gv13x2x-glibc-2.35-224-bin/bin:/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin:/nix/store/62fb427ncxaaksa2k59rhbilfg68v20x-binutils-wrapper-2.39/bin:/nix/store/frh9l9nrdysasdi2gs7i241s241ngjw2-binutils-2.39/bin:/nix/store/w73bxii8xlsm9459628glawfcjlci75d-gobject-introspection-1.74.0-dev/bin:/nix/store/h8gvq6r4hgpa71h44dmg9qfx03mj81sv-coreutils-9.1/bin:/nix/store/zml88vnkpm8if114qkbbqd1q7n3ypqqy-findutils-4.9.0/bin:/nix/store/49y3r0gr9m6k20d91kl6dgp4b9a6m72v-diffutils-3.8/bin:/nix/store/5dv5cq1lwvsijr9p16p2kp79g1dbajk3-gnused-4.8/bin:/nix/store/bcvccw6y9bfil6xrl5j7psza7hnd16ry-gnugrep-3.7/bin:/nix/store/l1fp0hyca54xbb85vfhppd16bskzx8dg-gawk-5.1.1/bin:/nix/store/89zbjdkb48ma61k76l2mzn3s0ra0wn2c-gnutar-1.34/bin:/nix/store/qs8qb1swpivkfq7i9yd52n0mw6z4ij81-gzip-1.12/bin:/nix/store/wwkyfg8b34xy16zzc9p6rkh59p4q37qx-bzip2-1.0.8-bin/bin:/nix/store/i0x4pzj96qwvkrm94317l6jbi53a2rdj-gnumake-4.4/bin:/nix/store/4xw8n979xpivdc46a9ndcvyhwgif00hz-bash-5.1-p16/bin:/nix/store/793iwfbjvg7wgpqq7r83a1qjl1yg02sf-patch-2.7.6/bin:/nix/store/a3mwv26f99ycsv9w6hrx0jjjjywvcb1n-xz-5.2.9-bin/bin:/nix/store/camlh5laf1wsklghk0vcaw7gvx4rpzd1-file-5.43/bin"
```

Reply:
if someone does this then the fake-uname won't be used but if coreutils is after fake-uname then the fake is used
```
diff --git a/pkgs/development/libraries/libgtop/default.nix b/pkgs/development/libraries/libgtop/default.nix
index 59db7929240..8a256e5f811 100644
--- a/pkgs/development/libraries/libgtop/default.nix
+++ b/pkgs/development/libraries/libgtop/default.nix
@@ -8,6 +8,7 @@
 , gnome
 , gtk-doc
 , fake-uname
+, coreutils
 }:
 
 stdenv.mkDerivation rec {
@@ -27,6 +28,7 @@ stdenv.mkDerivation rec {
     perl
     gettext
     gobject-introspection
+    coreutils
     fake-uname
   ];
 ```

Reply:
Or possibly when one of the preceding dependencies propagates it?